internal/resources/otelcollector/rules.go

```diff
@@ -0,0 +1,83 @@
+package otelcollector
+
+import rbacv1 "k8s.io/api/rbac/v1"
+
+var (
+
+	// policy rules needed for the k8sattributeprocessor component
+	k8sAttributeRules = []rbacv1.PolicyRule{{
```

Comment:
why define those rule outside the functional options?
do we need them anywhere else?